tensorbay/client/profile.py

```diff
@@ -121,6 +150,32 @@ def wrapper(client: Client, method: str, url: str, **kwargs: Any) -> Response:
 
         return wrapper  # type: ignore[return-value]
 
+    def _statistical_read(self, response: HTTPResponse, amt: Optional[int] = None) -> bytes:
+        start_time = time.time()
+        content = HTTPResponse.read(response, amt)
+        self._update(self._download_path, 0, time.time() - start_time, len(content))
+        return content
+
+    def _statistical_open(self, func: _OpenCallable) -> _OpenCallable:
+        @wraps(func)
+        def wrapper(obj: RemoteFileMixin) -> HTTPResponse:
+            if not hasattr(self, "download_path"):
+                netloc = urlparse(obj.url.get()).netloc  # type: ignore[union-attr]
+                self._download_path = f"[GET] {netloc}/*"
```

Comment:
what if we have multiple `_download_path`?

Reply:
Multiple `download_path` only occured between different dataset.
In the case about reading data from dataset, It seems hardly appears.
By the way, MethodType not support send other auguments.

Reply:
using `Profile` instance to share the `download_path` is weird, will use closure better?